### DISCUSSION TITLE
How do you figure out which file to load?

### DISCUSSION THREAD

**Discussion by babakfp** [fontsource/fontsource#985] [Q&A] [answered]
Hi 👋

I'm trying to preload normal and mono versions.

![](https://github.com/user-attachments/assets/19a7f902-6fcb-412d-8df1-54ed1591c37e)

😐

This is what I'm importing:

```js
import "@fontsource-variable/recursive"
import "@fontsource-variable/recursive/mono.css"
```

**Comment by ayuhito** [comment, marked as answer]
Your imports should look like this:

```ts
import "@fontsource-variable/recursive/mono.css"
import recursiveLatinWoff2 from "@fontsource-variable/recursive/files/recursive-latin-mono-normal.woff2?url";
```

This is assuming you have an English website, in which case the `latin` subset is the most appropriate file for you to preload. You should only preload one or two critical WOFF2 files as excessively preloading many files can have [negative effects](https://web.dev/articles/preload-critical-assets#effects_of_preloading_on_core_web_vitals).

You also only need to import one CSS file, not multiple as `mono` is a superset of the regular variable font import and has everything needed.

**Reply by babakfp** [reply]
Awesome, thanks!

> You also only need to import one CSS file, not multiple as `mono` is a superset of the regular variable font import and has everything needed.

Ahh, that's great to know 👍. Is this in the docs? I mean, I didn't fully read the docs but if it isn't there, it would be great if it was added.

Thanks again 🌸.

**Reply by babakfp** [reply]
Can you also explain the difference between these two, please? (one has `ext` in the name).

```
import recursive from "@fontsource-variable/recursive/files/recursive-latin-mono-normal.woff2"
import recursive from "@fontsource-variable/recursive/files/recursive-latin-ext-mono-normal.woff2"
```

**Reply by ayuhito** [reply]
> Ahh, that's great to know 👍. Is this in the docs? I mean, I didn't fully read the docs but if it isn't there, it would be great if it was added.

This is [explained in the docs](https://fontsource.org/docs/getting-started/variable), and we also have a convenient helper tool to simplify selecting imports on [the install page of every font](https://fontsource.org/fonts/recursive/install).

> Can you also explain the difference between these two, please? (one has ext in the name).

This [StackOverflow post](https://stackoverflow.com/questions/14303677/languages-supported-by-latin-vs-latin-extended-glyphs-in-fonts-on-google-web) explains it pretty well. These two do not overlap and contain completely different characters. If you needed both, you need to import both and not just `latin-ext`.

**Comment by babakfp** [comment]
I think I understand it now. These (`cyrillic`, `latin`, `vietnamese`) things are like groups of supported alphabets. So, if you want to use the font for English, you should use `latin`. Is there a documentation for this in the site? It would be great to have a little explanation at the end of https://fontsource.org/docs/getting-started/preload. I'm open to create a PR if needed.
For example, this is the wiki page about Cyrillic: https://en.wikipedia.org/wiki/Cyrillic_script

**Reply by ayuhito** [reply]
I'm careful about including much documentation when it comes to the individual subsets of the fonts because more often than not, developers misuse it (and have done so in the past) when the defaults are the most optimal for everyone. This was a big problem when we introduced [Individual Subsets](https://fontsource.org/docs/getting-started/subsets) which is only truly useful for some _niche_ use cases.

But I do agree that maybe a little bit more explanation on what the subsets would help for the preload docs. I wouldn't include the wiki pages since no is really going to read through that, but a brief explanation on what the subsets are to complement the blockquote at the bottom emphasizing only to import critical subsets would be pretty great to have!
